ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/misc.py

```diff
@@ -210,3 +210,11 @@ def isis_default_circuit_type(self: SharedUtils) -> str | None:
     @cached_property
     def pod_name(self: SharedUtils) -> str | None:
         return get(self.hostvars, "pod_name")
+
+    @cached_property
+    def fabric_sflow(self: SharedUtils) -> dict | None:
```

Comment:
This returns one dict which may contain one or more of the allowed keys in the schema. But it is up to the function consuming this, to detect which keys are set.
I would split this out into separate boolean properties for each key. So `fabric_sflow_uplinks(self: SharedUtils) -> bool | None:`
This way we can simplify the code on adapters etc. to just say 
```python
adapter['sflow'] = default(adapter_settings.get("sflow"), shared_utils.fabric_sflow_endpoints)
```
It will allow you to override either true or false both ways on the adapter, and we will return true and false values in `structured_config`.